### DISCUSSION TITLE
Cannot see big files

### DISCUSSION THREAD

**Discussion by hyperknot** [mitmproxy/mitmproxy#6417] [Q&A] [answered]
Hi, I'm trying to debug why one macOS application uses extreme amount of network traffic, like 8 MB/s in the background. It's using the WebDAV protocol, and I managed to set up the reverse proxy mode.

It works, however it only analyses / dumps tiny amount of data, like 1 MB, even though the app is transferring GBs of data.

I tried dumping with -w but the resulting file is just 1 MB. If I kill the app, the transfers terminate, so it has to go through the proxy somehow.

How can I make mitmproxy actually display/dump the full HTTP/HTTPS stream?

**Comment by hyperknot** [comment]
Is it as simple as `--set stream_large_bodies=100g`?

**Reply by hyperknot** [reply]
Is there a way to detect the size while it's still streaming?

**Comment by mhils** [comment, marked as answer]
By default, mitmproxy only operates on full/complete HTTP bodies. That means mitmproxy will download/cache an entire response, then process it, and only then forward it to the client. You can change that by setting `stream_large_bodies`, in which case bodies are forwarded immediately and not kept in memory.

See https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/http-stream-simple.py and https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/http-stream-modify.py for programmatic access. Large, single-request HTTP downloads unfortunately don't fit mitmproxy very well at the moment.

**Reply by hyperknot** [reply]
Thanks. The limit is memory? So a 30 MB file works well but a 10 GB file doesn't?
